tests/scripts/run-xunit-perf.cmd

```diff
@@ -2,193 +2,332 @@
+rem   Executes the xUnit Performance benchmarks
+rem ****************************************************************************
+setlocal
+  set BENCHNAME=%~n1
+  set BENCHDIR=%~p1
+  set PERFOUT=perf-%BENCHNAME%
+  set XMLOUT=%PERFOUT%.xml
+
+  rem copy benchmark and any input files
+  call :run_cmd xcopy /s %~1 . >> %RUNLOG%  || exit /b 1
+  if exist "%BENCHDIR%*.txt" (
+    call :run_cmd xcopy /s %BENCHDIR%*.txt . >> %RUNLOG%  || exit /b 1
+  )
+
+  set CORE_ROOT=%CORECLR_REPO%\sandbox
+
+  rem setup additional environment variables
+  if DEFINED TEST_ENV (
+    if EXIST "%TEST_ENV%" (
+        call "%TEST_ENV%"
+    )
+  )
+
+  set BENCHNAME_LOG_FILE_NAME=%BENCHNAME%.log
+  if defined IS_SCENARIO_TEST (
+    call :run_cmd corerun.exe "%CORECLR_REPO%\sandbox\%BENCHNAME%.%TEST_FILE_EXT%" --perf:runid Perf 1>"%BENCHNAME_LOG_FILE_NAME%" 2>&1
+  ) else (
+    call :run_cmd corerun.exe PerfHarness.dll "%CORECLR_REPO%\sandbox\%BENCHNAME%.%TEST_FILE_EXT%" --perf:runid Perf 1>"%BENCHNAME_LOG_FILE_NAME%" 2>&1
+  )
+
+  IF %ERRORLEVEL% NEQ 0 (
+    call :print_error corerun.exe exited with %ERRORLEVEL% code.
+    if exist "%BENCHNAME_LOG_FILE_NAME%" type "%BENCHNAME_LOG_FILE_NAME%"
+    exit /b 1
+  )
+
+  rem optionally generate results for benchview
+  if not [%BENCHVIEW_PATH%] == [] (
+    call :generate_results_for_benchview || exit /b 1
   ) else (
-    py "%BENCHVIEW_PATH%\measurement.py" xunit "Perf-%BENCHNAME%.xml" --better desc --drop-first-value --append
-  )
-  REM Save off the results to the root directory for recovery later in Jenkins
-  xcopy Perf-%BENCHNAME%*.xml %CORECLR_REPO%\
-  xcopy Perf-%BENCHNAME%*.etl %CORECLR_REPO%\
-) else (
-  type %XMLOUT% | findstr "test name"
-  type %XMLOUT% | findstr Duration
-  type %XMLOUT% | findstr InstRetired
-)
-
-goto :EOF
-
-:ARGLOOP
-IF /I [%1] == [-testBinLoc] (
-set CORECLR_PERF=%CORECLR_REPO%\%2
-shift
-shift
-goto :ARGLOOP
-)
-IF /I [%1] == [-scenarioTest] (
-set IS_SCENARIO_TEST=1
-shift
-goto :ARGLOOP
-)
-IF /I [%1] == [-runtype] (
-set RUN_TYPE=%2
-shift
-shift
-goto :ARGLOOP
-)
-IF /I [%1] == [-library] (
-set TEST_FILE_EXT=dll
-shift
-goto :ARGLOOP
-)
-IF /I [%1] == [-uploadtobenchview] (
-set BENCHVIEW_PATH=%2
-shift
-shift
-goto :ARGLOOP
-)
-IF /I [%1] == [-arch] (
-set TEST_ARCHITECTURE=%2
-shift
-shift
-goto :ARGLOOP
-)
-IF /I [%1] == [-testEnv] (
-set TEST_ENV=%2
-shift
-shift
-goto :ARGLOOP
-)
-IF /I [%1] == [-configuration] (
-set TEST_CONFIG=%2
-shift
-shift
-goto :ARGLOOP
-)
-if /I [%1] == [-?] (
-goto :USAGE
-)
-if /I [%1] == [-help] (
-goto :USAGE
-)
-
-if [%CORECLR_PERF%] == [] (
-goto :USAGE
-)
-
-goto :SETUP
+    type "%XMLOUT%" | findstr /i /c:"test name"
+  )
+
+  rem Save off the results to the root directory for recovery later in Jenkins
+  call :run_cmd xcopy "Perf-%BENCHNAME%*.xml" "%CORECLR_REPO%\" || exit /b 1
+  call :run_cmd xcopy "Perf-%BENCHNAME%*.etl" "%CORECLR_REPO%\" || exit /b 1
```

Comment:
You don't have to do this for this change, but it would be nice to copy over the %BENCHNAME_LOG_FILE_NAME% files (here), as well as %RUNLOG% (in the future), and archive them in addition to the etl and xml files (archiving would mean changes to perf.groovy), especially in the case of failed benchmarks.